aea/mail/base.py

```diff
@@ -248,6 +248,46 @@ def decode(self, envelope_bytes: bytes) -> "Envelope":
 class ProtobufEnvelopeSerializer(EnvelopeSerializer):
     """Envelope serializer using Protobuf."""
 
+    @staticmethod
+    def _get_protocol_specification_id_by_the_protocol_id(
+        protocol_id: PublicId,
+    ) -> PublicId:
+        """
+        Get protocol specification id  from protocol id.
+
+        If not protocol was registered with given specification id, returns protocol specification id  itself.
+
+        :param protocol_id: PublicId
+
+        :return: PublicId
+        """
+        return (
+            ProtocolSpecificationsRegistry.get_specification_id_by_protocol_id(
+                protocol_id
+            )
+            or protocol_id
+        )
+
+    @staticmethod
+    def _get_protocol_id_by_protocol_specification_id(
+        protocol_specification_id: PublicId,
+    ) -> PublicId:
+        """
+        Get protocol id from specification id.
+
+        If not protocol was registered with given specification id, returns protocol_id itself.
+
+        :param protocol_specification_id: PublicId
+
+        :return: PublicId
+        """
+        return (
+            ProtocolSpecificationsRegistry.get_protocol_id_by_specification_id(
+                protocol_specification_id
+            )
+            or protocol_specification_id
+        )
```

Comment:
The envelope should not be concerned with this. This needs to be happening outside its scope

Reply:
In particular, look at where we call encode and decode on envelopes from. That's where we need to do the translation of the ids.